### PR TITLE
feat: download resources on the fly

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

File: src/downloader.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    'linux',
+    'x64',
+    true,
+  )
+  await ensureAsset(
+    'https://github.com/ethersphere/bee/releases/download/v1.5.1/bee-windows-amd64.exe',
+    'bee.exe',
+    'win32',
+    'x64',
+  )
```
Comment (vojtechsimetka) [inline src/downloader.ts:43]: I would suggest to put the platform check here as a `switch` statement. It would make it more readable.

Comment (vojtechsimetka) [reply src/downloader.ts:43]: the architecture could then be an if statement.

**Review by vojtechsimetka — APPROVED** [review]
I like this way better. Just a few observation:
- the downloaded zip files are not erased (e.g. the asset zip file)
- the app does not start until all the assets are downloaded. This is confusing, IMHO we should show some window saying that the assets are being downloaded
- it now stores everything in `/Users/<user>/Library/Application Support/bee-desktop-nodejs` which I personally like a lot more, but we should provide a way to set this path through env variable (can be done in a different PR)

**Comment by Cafe137** [comment]
@vojtechsimetka Can you please review downloader.ts?

**Review by vojtechsimetka — APPROVED** [review]
Oh yeah thank you, this is much better!

**Pull request merged**